gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcLoggingInterceptor.java

```diff
@@ -0,0 +1,118 @@
+package com.google.api.gax.grpc;
+
+import com.google.api.gax.logging.LoggingUtils;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import io.grpc.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.event.Level;
+
+public class GrpcLoggingInterceptor implements ClientInterceptor {
+
+  private static final Logger logger = LoggingUtils.getLogger(GrpcLoggingInterceptor.class);
+  private static final Gson gson = new Gson();
+
+  @Override
+  public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+      MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
+
+    Map<String, String> serviceAndRpc = new HashMap<>();
+    Map<String, String> requestLogData = new HashMap<>();
+    Map<String, String> responseLogData = new HashMap<>();
+
+    // Initialize a JsonArray to hold all responses
+    JsonArray responsePayloads = new JsonArray();
+    return new ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(
+        next.newCall(method, callOptions)) {
+
+      @Override
+      public void start(Listener<RespT> responseListener, Metadata headers) {
+        if (logger.isInfoEnabled()) {
+          String requestId = UUID.randomUUID().toString();
```

Comment:
I think `requestId` is for debugging purpose for now and would be removed later?

Reply:
Yes for now. Later when we also log in tracer, we may need some requestID to track (not this one).